src/browser-frontend/sagas/server-event-handlers.js

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import { takeEvery, call, put, select } from 'redux-saga/effects';
+
+import SharedActions from '../../shared/actions/shared-actions';
+import PagesModelActions from '../actions/pages-model-actions';
+import * as DomainPagesSelectors from '../selectors/domain-pages-selectors';
+import * as UIPagesSelectors from '../selectors/ui-pages-selectors';
+
+function* handleCmdOrControlQ({ meta: client }) {
```

Comment:
Just as a note, we could follow Firefox's conventions, and call it `handleAccelQ`…  😉
(Actually, we can't because of electron-localshortcut.  But I would prefer the shorter `CmdOrCtrl+Q`.)

Also, your long note should probably be a comment somewhere.  Either in the code, or in the architecture document would work for me.